src/component/editor/CodeEditorComponent.tsx

```diff
@@ -1,85 +1,42 @@
 import React from 'react';
-import 'codemirror/lib/codemirror.css';
-import 'codemirror/addon/lint/lint.css';
-import 'codemirror/addon/hint/show-hint.css';
-import CypherEditor from './CypherEditor';
+import { CypherEditor, CypherEditorProps } from '@neo4j-cypher/react-codemirror';
+import { markdown, markdownLanguage } from '@codemirror/lang-markdown';
+// import { languages } from '@codemirror/language-data';
+
+const markdownExtensions = [
+  markdown({
+    base: markdownLanguage, // Support GFM
+    // codeLanguages: languages
+  }),
+];
+
+import '@neo4j-cypher/codemirror/css/cypher-codemirror.css';
 
 const NeoCodeEditorComponent = ({
   value,
-  onChange = () => {},
+  onChange,
   placeholder,
   editable = true,
   language = 'cypher',
   style = { width: '100%', height: 'auto', border: '1px solid lightgray' },
 }) => {
-  const [cancelNextEdit, setCancelNextEdit] = React.useState(false);
-  const options = {
-    viewPortMargin: Infinity,
-    mode: language,
-    theme: 'cypher',
-    height: 'auto',
-    lineNumberFormatter: (line) => line,
+  const editorProps: CypherEditorProps = {
+    cypherLanguage: language === 'cypher',
+    readOnly: !editable,
+    placeholder: placeholder,
+    preExtensions: language === 'markdown' ? markdownExtensions : [],
```

Comment:
Markdown doesn't appear to be working for me:
![image](https://user-images.githubusercontent.com/1754968/210564161-1995e232-0fd3-4790-8fe4-757db6a2a82c.png)

Reply:
It is working, it just doesn't show any markdown styling unless we add a theme extension which I didn't do. It's subtle but you can see that the markdown extension is working because it does things like auto insert next list item number etc.

That said, we should probably have a chat about what markdown syntax/theme to use.